src/wire/parseRefsAdResponse.js

```diff
@@ -41,14 +41,17 @@ export async function parseRefsAdResponse(stream, { service }) {
 
   const [firstRef, capabilitiesLine] = splitAndAssert(lineTwo, '\x00', '\\x00')
   capabilitiesLine.split(' ').map(x => capabilities.add(x))
-  const [ref, name] = splitAndAssert(firstRef, ' ', ' ')
-  refs.set(name, ref)
-  while (true) {
-    const line = await read()
-    if (line === true) break
-    if (line !== null) {
-      const [ref, name] = splitAndAssert(line.toString('utf8'), ' ', ' ')
-      refs.set(name, ref)
+  // see no-refs in https://git-scm.com/docs/pack-protocol#_reference_discovery (since git 2.41.0)
+  if (firstRef !== '0000000000000000000000000000000000000000 capabilities^{}') {
+    const [ref, name] = splitAndAssert(firstRef, ' ', ' ')
+    refs.set(name, ref)
+    while (true) {
+      const line = await read()
+      if (line === true) break
+      if (line !== null) {
+        const [ref, name] = splitAndAssert(line.toString('utf8'), ' ', ' ')
+        refs.set(name, ref)
+      }
```

Comment:
I believe this is from another commit. Is it worth a rebase or merge to keep it separate?

Reply:
Yea, it was the commit to handle empty repositories.

Reply:
I will need to redo the merge locally, because GitHub required to squash the whole history.

Reply:
Created second PR https://github.com/isomorphic-git/isomorphic-git/pull/1902